variables.tf

```
@@ -0,0 +1,85 @@
+variable "namespace" {
+  description = "Namespace (e.g. `eg` or `cp`)"
+  type        = "string"
+}
+
+variable "stage" {
+  description = "Stage (e.g. `prod`, `dev`, `staging`)"
+  type        = "string"
+}
+
+variable "name" {
+  description = "Name  (e.g. `app` or `db`)"
+  type        = "string"
+}
+
+variable "delimiter" {
+  type        = "string"
+  default     = "-"
+  description = "Delimiter to be used between `namespace`, `stage`, `name` and `attributes`"
+}
+
+variable "attributes" {
+  type        = "list"
+  default     = []
+  description = "Additional attributes (e.g. `1`)"
+}
+
+variable "tags" {
+  type        = "map"
+  default     = {}
+  description = "Additional tags (e.g. map(`BusinessUnit`,`XYZ`)"
```

```suggestion
  description = "Additional tags (e.g. `{ BusinessUnit = \"XYZ\" }`"
```

I think @aknysh recommend the backticks, but that's not valid terraform. Also the `map(...)` is an artifact from more than a year ago (maybe 2) when terraform didn't support the `{ ... }` notation. Let's move to map notation.